python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets_job.py

```diff
@@ -1649,3 +1619,57 @@ def my_derived_asset(my_source_asset):  # pylint: disable=unused-argument
         match="resource with key 'foo' required by resource with key 'my_source_asset__io_manager' was not provided.",
     ):
         build_assets_job(name="test", assets=[my_derived_asset], source_assets=[my_source_asset])
+
+
+def test_resolve_dependency_in_group():
+    @asset(key_prefix="abc")
+    def asset1():
+        ...
+
+    @asset
+    def asset2(context, asset1):
+        del asset1
```

Comment:
why is `del` necessary here?

Reply:
It's an unused argument - following the top-upvoted answer on this stackoverflow question: https://stackoverflow.com/questions/10025680/how-can-i-denote-unused-function-arguments